Mage.Sets/src/mage/cards/a/AminatousAugury.java

```diff
@@ -51,71 +56,98 @@ public AminatousAugury(final AminatousAugury card) {
-            TargetCard target = new TargetCard(
-                    Zone.EXILED,
-                    StaticFilters.FILTER_CARD_LAND_A
-            );
-            if (cardsToCast.count(StaticFilters.FILTER_CARD_LAND, game) > 0) {
-                if (controller.chooseUse(Outcome.PutLandInPlay, "Put a land from among the exiled cards into play?", source, game)) {
-                    if (controller.choose(Outcome.PutLandInPlay, cardsToCast, target, game)) {
-                        Card card = cardsToCast.get(target.getFirstTarget(), game);
-                        if (card != null) {
-                            cardsToCast.remove(card);
-                            controller.moveCards(card, Zone.BATTLEFIELD, source, game, false, false, true, null);
+
+            //Exile the top eight cards of your library.
+            Set<Card> cards = player.getLibrary().getTopCards(game, 8);
+            UUID exileId = CardUtil.getCardExileZoneId(game, source);
+
+
+            for (Card card : cards) {
+                card.moveToExile(exileId,source.getSourceObject(game).getName(), source.getSourceId(), game);
+                cardIdToAminatouIdMap.put(card.getId(), source.getSourceId());
+             }
+
+            game.getExile().getExileZone(exileId).setCleanupOnEndTurn(true);
+            game.getState().setValue("cardToAminatouIdMap", cardIdToAminatouIdMap);
+
+
+            //you may put a land card from among them onto the battlefield
+            Cards aminatouExileGroup = new CardsImpl(cards);
+            if(aminatouExileGroup.getCards(StaticFilters.FILTER_CARD_LAND, source.getSourceId(), source.getControllerId(), game).size() > 0 ){
+                if (player.chooseUse(Outcome.PutCardInPlay, "Put a land from among exiled cards on to the battlefield?", source, game)) {
+                    TargetCard targetLandCard = new TargetCard(1, Zone.EXILED, StaticFilters.FILTER_CARD_LAND);
+                    if (player.choose(Outcome.PutCardInPlay, aminatouExileGroup, targetLandCard, game)) {
+                        Card landCard = game.getCard(targetLandCard.getFirstTarget());
+                        if (landCard != null) {
+                            player.moveCards(landCard, Zone.BATTLEFIELD, source, game, false, false, false, null);
+                            aminatouExileGroup.remove(landCard);
                         }
                     }
                 }
             }
-            for (Card card : cardsToCast.getCards(StaticFilters.FILTER_CARD_NON_LAND, game)) {
-                AminatousAuguryCastFromExileEffect effect = new AminatousAuguryCastFromExileEffect();
-                effect.setTargetPointer(new FixedTarget(card.getId(), card.getZoneChangeCounter(game)));
-                game.addEffect(effect, source);
+
+            //create list of unused card types
+            EnumSet<CardType> unusedCardTypes = EnumSet.allOf(CardType.class);
+            unusedCardTypes.remove(CardType.LAND);
```

Comment:
I recommends save `used` card types instead all of them. Logic like that:
* Little amount of store data;
* You can use it to show extra info like `ValueHint` in animo card;
* Code more easy:
  * load used card types;
  * compare current card types with used list (like `findUnusedList(card, usedTypes)`;
  * if it contains unused then make able to cast and able to pay.